Object Detection Train

Dashrekker opened the pull request:
Implemented object detection trainspace. 
![image](https://github.com/DSGT-DLP/Deep-Learning-Playground/assets/47958374/c2b9fb64-44d3-4c01-b1ae-c48d0009a054)
![image](https://github.com/DSGT-DLP/Deep-Learning-Playground/assets/47958374/9f8d9e81-d5a4-428f-aa59-0f1635ff9bd5)

Dashrekker committed:
implemented object_detection_parameters and added object detection files

Dashrekker committed:
removed detect from navbar, updated trainconstants, and added detect parameters to trainspace_db

Dashrekker committed:
Merge branch 'nextjs' of https://github.com/DSGT-DLP/Deep-Learning-Playground into object-detection-train

Dashrekker committed:
removed playground-frontend and moved object detection

Dashrekker committed:
added renderstepperbuttons to detectionimagestep

Dashrekker committed:
added image upload layout file for filerobot implementation

Dashrekker committed:
converted transforms panel to a grid and implemented detectiontype state for toggling problemtypes

Dashrekker committed:
Merge branch 'nextjs' of https://github.com/DSGT-DLP/Deep-Learning-Playground into object-detection-train

Dashrekker committed:
fixed float not supported issue with update dynamo item and modified detection endpoint for trainspace

Dashrekker committed:
:art: Format Python code with psf/black

farisdurrani commented on backend/endpoints/train_endpoints.py:
```diff
@@ -57,7 +55,10 @@ def tabular_run():
     try:
         request_data = json.loads(request.data)
+        print("hello")
+        print(request_data)
```
Can you please remove unnecessary prints?

farisdurrani commented on frontend/src/common/components/NavBarMain.tsx:
```diff
@@ -83,90 +84,120 @@ const NavbarMain = () => {
-                          <Typography sx={{fontFamily:'Lato, Arial, Helvetica, sans-serif'}}>Log out</Typography>
+                          <Typography
+                            sx={{
+                              fontFamily: "Lato, Arial, Helvetica, sans-serif",
```
This is supposed to be declared globally, in globals.css, which I think we've done. So you can remove this line

Dashrekker committed:
removed prints and changed imagedata to datasetdata

Dashrekker committed:
fixed eslint errors

karkir0003 commented on frontend/src/features/Train/features/Tabular/components/TabularReviewStep.tsx:
```diff
@@ -23,7 +23,7 @@ const TabularReviewStep = ({
   );
   const {
     handleSubmit,
-    formState: { errors, isDirty },
```
@dwu359, is the `isDirty` variable planned to be in use here. eslint complained that `isDirty` is unused var, so we assumed no use case for this variable. Correct me if I'm wrong?

dwu359 replied:
Yeah I don't think that variable is being used, it says that on my end too

karkir0003 replied:
so can we remove it @dwu359?

Dashrekker committed:
fixed eslint errors

Dashrekker committed:
removed font style from navbar

Dashrekker committed:
Merge branch 'nextjs' of https://github.com/DSGT-DLP/Deep-Learning-Playground into object-detection-train

farisdurrani commented:
lgtm

Dashrekker committed:
changed problem_type to detection_type in detectionapi

Dashrekker committed:
Merge branch 'nextjs' of https://github.com/DSGT-DLP/Deep-Learning-Playground into object-detection-train

dwu359 reviewed (changes requested):
I received an error that looked something like this when spinning up frontend: 
```
import FilerobotImageEditor from"./components/AssemblyPoint";import{TABS_IDS,TOOLS_IDS}from"./utils/constants";export{TABS_IDS as TABS,TOOLS_IDS as TOOLS};export default FilerobotImageEditor;
^^^^^^

SyntaxError: Cannot use import statement outside a module
```

It may be because of the React Filerobot Image Editor not supporting nextjs directly. Try using a dynamic import: https://github.com/scaleflex/filerobot-image-editor/issues/228

dwu359 commented on frontend/src/features/Train/components/ImageUploadLayout.tsx:
```diff
@@ -0,0 +1,190 @@
+      <Stack direction={"row"} spacing={2}>
+        <FilerobotImageEditor
+          source="https://scaleflex.airstore.io/demo/stephen-walker-unsplash.jpg"
+          onSave={(editedImageObject: any) => {
```
Use unknown if that works instead of any

Dashrekker replied:
Changed the any to unknown

Dashrekker committed:
changed any to unknown

Dashrekker committed:
implemented dynamic imports on imageupload page

dwu359 commented on frontend/src/features/Train/features/ObjectDetection/constants/detectionConstants.ts:
```diff
@@ -0,0 +1,232 @@
+        parameters: [
+          {
+            label: "probability",
+            index: 0,
```
No need for index attribute, it's already part of a list

dwu359 commented on frontend/src/features/Train/features/ObjectDetection/constants/detectionConstants.ts:
```diff
@@ -0,0 +1,232 @@
+        parameters: [
+          {
+            label: "probability",
+            index: 0,
```
Same here (+ other places too in this file)

dwu359 commented on frontend/src/features/Train/features/ObjectDetection/constants/detectionConstants.ts:
```diff
@@ -0,0 +1,232 @@
+          {
+            label: "probability",
+            index: 0,
+            parameter_name: "prob",
```
What is parameter_name for btw? It doesn't seem to be referenced in your object detection code

Dashrekker replied:
I think its just a parameter from old components that I wasn't sure whether to remove or not. It does the same thing as label so I've removed it.

dwu359 commented on frontend/src/features/Train/features/ObjectDetection/constants/detectionConstants.ts:
```diff
@@ -0,0 +1,232 @@
+      },
+    
+      ToTensor: {
+        display_name: "To Tensor",
```
Label & display_name seem to be intended to be used in the same way. Since other constants files (imageConstants, tabularConstants) also use label instead of display_name, use label instead for consistency purposes

dwu359 commented on frontend/src/features/Train/features/ObjectDetection/constants/detectionConstants.ts:
```diff
@@ -0,0 +1,232 @@
+      AdjustContrast: {
+        display_name: "AdjustContrast",
+        objectName: "transforms.functional.adjust_contrast",
+        transform_type: "functional",
```
Use camelCase for object properties for consistency purposes. Also, what is transform_type used for?

Dashrekker replied:
Pytorch has a category of transforms called functional transforms. In the old implementation, I used this parameter so that the backend could recognize what type of transform needed to be used.

Dashrekker replied:
I removed it for now

dwu359 commented on frontend/src/features/Train/components/ImageUploadLayout.tsx:
```diff
@@ -0,0 +1,207 @@
+          savingPixelRatio={0}
+          previewPixelRatio={0}
+        />
+        <Button variant="outlined" onClick={() => refetch()}>
```
Refresh button should be positioned so that it is below the image editor instead of to the right of it

dwu359 commented on frontend/src/features/Train/components/ImageUploadLayout.tsx:
```diff
@@ -0,0 +1,207 @@
+  }
+);
+
+const TABS = dynamic(
```
TABS and TOOLS don't have to be dynamically imported (seems to give a weird typing error when it is)

Dashrekker replied:
I still get the import error when I import it normally (it happens when I refresh). Looking for a way to resolve the weird type error, but it doesn't seem to cause any issues when I run it.

dwu359 commented on frontend/src/features/Train/components/ImageUploadLayout.tsx:
```diff
@@ -0,0 +1,207 @@
+      <Stack direction={"row"} spacing={2}>
+        <FilerobotImageEditor
+          source="https://scaleflex.airstore.io/demo/stephen-walker-unsplash.jpg"
+          onSave={(editedImageObject: unknown) => {
```
No need to specify the type for the editedImageObject argument btw, typescript automatically infers it for you since the onSave property already specifies it

dwu359 reviewed (commented):
Overall, looks pretty good! Just a few minor things that are still there. Is there any way for the user to upload images to edit in the image editor in the dataset step, or does that image editor only work for the one image with mountains that it defaults to?

Dashrekker committed:
fixed type and refresh

Dashrekker committed:
removed extra stuff from transforms

Dashrekker committed:
removed transform type

Dashrekker commented:
@dwu359 I forgot to mention this earlier but you can upload images to edit and the mountain image is just a default backdrop. As for changing the mountain image, in the past I've changed it to other images such as a blank image but it was buggy so I changed it back.

Dashrekker committed:
Merge branch 'nextjs' of https://github.com/DSGT-DLP/Deep-Learning-Playground into object-detection-train

karkir0003 commented:
@dwu359 austin is working on integratimg tui image editor in place of file robot

Dashrekker committed:
implemented toastui with toastui wrapper class

Dashrekker committed:
Merge branch 'nextjs' of https://github.com/DSGT-DLP/Deep-Learning-Playground into object-detection-train

Dashrekker committed:
changed editor ref to ref and added functionality to toast ui

Dashrekker committed:
fixed linter errors and uninstalled filerobot

Dashrekker committed:
Merge branch 'nextjs' of https://github.com/DSGT-DLP/Deep-Learning-Playground into object-detection-train

farisdurrani commented:
Can you update the screenshot @Dashrekker? I think tui uses a dark background color, so that screenshot is old

karkir0003 commented:
> I received an error that looked something like this when spinning up frontend:
> 
> ```
> import FilerobotImageEditor from"./components/AssemblyPoint";import{TABS_IDS,TOOLS_IDS}from"./utils/constants";export{TABS_IDS as TABS,TOOLS_IDS as TOOLS};export default FilerobotImageEditor;
> ^^^^^^
> 
> SyntaxError: Cannot use import statement outside a module
> ```
> 
> It may be because of the React Filerobot Image Editor not supporting nextjs directly. Try using a dynamic import: [scaleflex/filerobot-image-editor#228](https://github.com/scaleflex/filerobot-image-editor/issues/228)

@dwu359 should be resolved. @Dashrekker removed the react-filerobot-image-editor deps

dwu359 reviewed (commented):
Looks good! I found uploading an image to be a little confusing though. When the user loads an image into the image editor, could you add an additional option to directly upload a file through the user's filesystem? The filename & jpg/png dropdown are probably not needed either, since that can automatically be inferred from either the image editor or thru the user's filesystem.
